### PR TITLE
add _supportedLocales in EasyLocalizationController; check the device…

### PR DESCRIPTION
**Pull request by bbsmartc**
add _supportedLocales in EasyLocalizationController; check the deviceLocale in resetLocale; 
--
to fix this [issue](https://github.com/aissat/easy_localization/issues/670)

add scriptCode if useOnlyLangCode
--
scriptCode is needed sometimes, for example zh-Hans, zh-Hant

### CODE REVIEW

File: lib/src/easy_localization_controller.dart [hunk 1]
```diff
@@ -205,7 +208,8 @@ class EasyLocalizationController extends ChangeNotifier {
   Future<void> resetLocale() async {
     EasyLocalization.logger('Reset locale to platform locale $_deviceLocale');
```
Comment (bw-flagship) [inline lib/src/easy_localization_controller.dart]: i would log locale instead of _deviceLocale here

**Review by bw-flagship — CHANGES_REQUESTED** [review]

**Review by bw-flagship — COMMENTED** [review]
oh and please add this to the changelog

**Comment by bbsmartc** [comment]
changelog added

**Comment by bbsmartc** [comment]
> oh and please add this to the changelog

changed added

**Comment by bw-flagship** [comment]
@bbsmartc  did you see my other comment?

**Comment by bbsmartc** [comment]
> @bbsmartc did you see my other comment?

sorry, forgot! in the latest pull request, the log is already modified to
```
    EasyLocalization.logger('Reset locale to $locale while the platform locale is $_deviceLocale and the fallback locale is $_fallbackLocale');

```

**Review by bw-flagship — APPROVED** [review]

**Pull request merged**